Baikal/PostEffects/ML/buffer.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <vector>
+#include <cstddef>
+#include <memory>
+
+
+namespace Baikal
+{
+    namespace PostEffects
+    {
+        class Buffer
+        {
+        public:
+            using ValueType = float;
+            using Data = std::unique_ptr<ValueType[]>;
+            using Shape = std::tupel<size_t, size_t, size_t>;
+
+            Buffer(Data data, Shape shape) :
+                m_data(std::move(data)),
+                m_shape(std::move(shape)),
+                m_size(std::get<0>(m_shape) *
+                       std::get<1>(m_shape) *
+                       std::get<2>(m_shape))
+            {
+            }
+
+            bool empty() const
+            {
+                return m_size == 0;
```

Comment:
size is always  known, because Inference knows it. We should change check here

Reply:
There was an idea to make shape (0, 0, 0), but it sounds like it is better to check nullptr for data.

Reply:
yes, it's better